File: javascript/jssearch/tests/searchfile.test.js
/*
 * searchresult_test.js
 *
 * Some tests of searchresult.js
 */

const {FileType} = require('../src/filetype');
const SearchFile = require('../src/searchfile').SearchFile;

describe('testing searchfile', () => {
    it('testSearchFileAbsPath', () => {
        const pathname = '/Users/cary/src/xsearch/javascript/jssearch/src';
        const filename = 'searchfile.js';
        const searchfile = new SearchFile(pathname, filename, FileType.CODE);
        const expected = '/Users/cary/src/xsearch/javascript/jssearch/src/searchfile.js';
        expect(searchfile.relativePath()).toEqual(expected);
    });

    it('testSearchFileRelPath1', () => {
        const pathname = '.';
        const filename = 'searchfile.js';
        const searchfile = new SearchFile(pathname, filename, FileType.CODE);
        const expected = './searchfile.js';
        expect(searchfile.relativePath()).toEqual(expected);
    });

    it('testSearchFileRelPath2', () => {
        const pathname = './';
        const filename = 'searchfile.js';
        const searchfile = new SearchFile(pathname, filename, FileType.CODE);
        const expected = './searchfile.js';
        expect(searchfile.relativePath()).toEqual(expected);
    });

    it('testSearchFileRelPath3', () => {
        const pathname = '..';
        const filename = 'searchfile.js';
        const searchfile = new SearchFile(pathname, filename, FileType.CODE);
        const expected = '../searchfile.js';
        expect(searchfile.relativePath()).toEqual(expected);
    });
});

File: cpp/cppsearch/tests/SearchFileTest.cpp
#include <catch2/catch.hpp>
#include "SearchFile.h"

TEST_CASE("Verify fullpath search file string equals expected", "[SearchFile]") {
    std::string path = "/Users/cary/src/xsearch/cpp/cppsearch/src";
    std::string file_name = "Searcher.cpp";
    auto *search_file = new cppsearch::SearchFile(path, file_name, cppsearch::FileType::CODE);

    REQUIRE(search_file->path() == "/Users/cary/src/xsearch/cpp/cppsearch/src");
    REQUIRE(search_file->file_name() == "Searcher.cpp");
    REQUIRE(search_file->file_type() == cppsearch::FileType::CODE);
    REQUIRE(search_file->string() == "/Users/cary/src/xsearch/cpp/cppsearch/src/Searcher.cpp");
}

TEST_CASE("Verify relative path search file string equals expected", "[SearchFile]") {
    std::string path = ".";
    std::string file_name = "Searcher.cpp";
    auto *search_file = new cppsearch::SearchFile(path, file_name, cppsearch::FileType::CODE);

    REQUIRE(search_file->path() == ".");
    REQUIRE(search_file->file_name() == "Searcher.cpp");
    REQUIRE(search_file->file_type() == cppsearch::FileType::CODE);
    REQUIRE(search_file->string() == "./Searcher.cpp");
}
